### PR TITLE
Topic 2 up to ex 8

### PR DESCRIPTION
**Pull request by macasaenz**

### CODE REVIEW

File: maca/02-oop-inheritance/classes.js [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+      });
+    }
+    else{
+      this.cast.push(cast); 
+    }
+
+  }
+
+  //adds array as an element 
+  addCast2(cast){
```
Comment (marcelososa) [inline maca/02-oop-inheritance/classes.js]: This method do the same that `addCast`, what is its purpose?

File: maca/02-oop-inheritance/classes.js [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+
+//Exercise 3
+/**
+ * Determines if an object is a function
+ * 
+ * @return {Boolean}
+ */
+let isFunction = function(obj) {  
+    return typeof obj == 'function' || false;
```
Comment (marcelososa) [inline maca/02-oop-inheritance/classes.js]: `typeof obj == 'function'` just return true or false

File: maca/02-oop-inheritance/classes.js (hunk 1 above)
Comment (macasaenz) [inline maca/02-oop-inheritance/classes.js]: It was just another option, the first one I thought of. It works ok with a single actor, but when adding an array of actors, it adds the whole array as one element. In addCast when it adds an array of actors it goes through it and adds each actor. 
I don't know if im being clear, but basically its the same function, just different ways of adding the actors. I think addCast is better, but I left this one just to show another option. 
Should I delete it?

File: maca/02-oop-inheritance/classes.js (hunk 2 above)
Comment (macasaenz) [inline maca/02-oop-inheritance/classes.js]: Done

**Comment by marcelososa** [comment]
Changes looks good. 

I'm so sorry, I forgot to say that you need to split `classes.js` file in different modules, at least one file per class and create a final bundle with Babel.

Could you do it?

**Comment by macasaenz** [comment]
I did it, but when i did the transpiling it created an empty file, and i
couldnt find a solution.. then i tried again in a different folder (i
realized then that ive been working in the C directory, dont know why, and
Matthieu suggested that could be a problem) and it worked, but when i
executed the bundle file created, i got an error im still trying to figure
out.

Ill try to have it done tommorrow, and if I can't I'll send an email to the
group asking for help, if that's ok..?

2017-04-12 21:11 GMT-03:00 Marcelo Sosa <notifications@github.com>:

**Comment by macasaenz** [comment]
When I execute index.html i get this error: 

Test exercise 4                                          bundle.js:274 
Uncaught ReferenceError: pururu is not defined              bundle.js:153
    at new Movie (bundle.js:153)
    at testEx4 (bundle.js:275)
    at window.onload (bundle.js:255)

That function is defined, but in the bundle it's defined at the end, I suppose that's the problem...? I don't know if i'm doing the bundle/modules wrong, or if i can't leave those functions like that, or if i can just reorganize the bundle file....?

**Pull request merged**